### PR TITLE
Development vdc rest interface

### PR DESCRIPTION
**Pull request by ranatrk**
### Description

Create a REST interface to be used with the current vdc dashboard

### Changes
- Add controller bottle server (REST interface endpoints) in vdc dashboard package
-  Create a common vdc helpers file to include common functionalities between bottle servers in vdcc dashboard
- Add controller_autherized decorator in auth
- Add delete button on storage nodes in vdc dashboard ui

### Related Issues
- https://github.com/threefoldtech/js-sdk/issues/2743
- https://github.com/threefoldtech/js-sdk/issues/2774#event-4468740241

### Checklist

- [x] [Pre-commit hook is installed](https://github.com/threefoldtech/js-ng#pre-commit) to do formatting checks before committing code...etc
- [ ] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstrings

### CODE REVIEW

File: jumpscale/packages/vdc_dashboard/bottle/controller.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+from jumpscale.packages.auth.bottle.auth import SESSION_OPTS, controller_authorized
+from jumpscale.packages.vdc_dashboard.bottle.vdc_helpers import get_vdc, threebot_vdc_helper
+from jumpscale.sals.vdc.size import VDC_SIZE
+
+
+app = Bottle()
+
+
+def _get_vdc_dict():
+    vdc = get_vdc()
```
Comment (waleedhammam) [inline jumpscale/packages/vdc_dashboard/bottle/controller.py:15]: do we need to vdc.load info() ?
I think kubernetes and 3bot data isn't filled at the begining

File: jumpscale/packages/vdc_dashboard/bottle/controller.py [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+        abort(400, "Error: Not all required params were passed.")
+
+    vdc = get_vdc()
+    vdc.load_info()
+    deployer = vdc.get_deployer(password=vdc_password)
+
+    try:
+        deployer.delete_k8s_node(wid)
+    except j.exceptions.Input:
+        abort(400, "Error: Failed to delete workload")
```
Comment (waleedhammam) [inline jumpscale/packages/vdc_dashboard/bottle/controller.py:126]: I think many exceptions could happen here

```suggestion
    except Exception as e:
        j.logger.error("Error: Failed to delete workload due to {str(e)}")
        abort(500, "Error: Failed to delete workload")
```

File: jumpscale/packages/vdc_dashboard/bottle/controller.py [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    if not farm:
+        zdb_farms = zdb_monitor.get_zdb_farm_names()
+        farm = random.choice(zdb_farms)
+    try:
+        wids = zdb_monitor.extend(required_capacity=capacity, farm_names=[farm], wallet_name="prepaid_wallet")
+        return HTTPResponse(
+            j.data.serializers.json.dumps(wids), status=201, headers={"Content-Type": "application/json"}
+        )
+    except Exception as e:
+        abort(500, f"Error: Failed to deploy zdb due to the following {str(e)}")
```
Comment (waleedhammam) [inline jumpscale/packages/vdc_dashboard/bottle/controller.py:183]: I think we can throw 500 but not with the exception details, alternatively we can log error it, to encapsulate the errors for our side

File: jumpscale/packages/vdc_dashboard/bottle/controller.py [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+        abort(400, "Error: Not all required params were passed.")
+
+    vdc = get_vdc()
+    vdc.load_info()
+    deployer = vdc.get_deployer(password=vdc_password)
+
+    try:
+        deployer.delete_s3_zdb(wid)
+    except j.exceptions.Input:
+        abort(400, "Error: Failed to delete workload")
```
Comment (waleedhammam) [inline jumpscale/packages/vdc_dashboard/bottle/controller.py:210]: same as last exception

File: jumpscale/packages/vdc_dashboard/bottle/deployments.py [hunk 5]
```diff
@@ -136,10 +124,28 @@ def delete_node():
...
+    if not wid:
+        abort(400, "Error: Not all required params was passed.")
+    vdc = get_vdc()
+    if not vdc:
+        return HTTPResponse(status=404, headers={"Content-Type": "application/json"})
+    deployer = vdc.get_deployer()
+    try:
+        deployer.delete_s3_zdb(wid)
+    except j.exceptions.Input:
+        abort(400, "Error: Failed to delete workload")
```
Comment (waleedhammam) [inline jumpscale/packages/vdc_dashboard/bottle/deployments.py:107]: exception maybe like before

File: jumpscale/packages/vdc_dashboard/bottle/vdc_helpers.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                addons.append(addon)
+    return addons
+
+
+def _total_capacity(vdc):
+    vdc.load_info()
+    addons = _get_addons(vdc.flavor, vdc.kubernetes)
+    plan = VDC_SIZE.VDC_FLAVORS.get(vdc.flavor)
+    plan_nodes_count = plan.get("k8s").get("no_nodes")
+    # total capacity = worker plan nodes + added nodes + master node
```
Comment (waleedhammam) [inline jumpscale/packages/vdc_dashboard/bottle/vdc_helpers.py:37]: commented code ?

File: jumpscale/packages/vdc_dashboard/bottle/controller.py (hunk 1 above)
Comment (ranatrk) [reply jumpscale/packages/vdc_dashboard/bottle/controller.py:15]: get_vdc() itself passes `load_info=True`

**Review by waleedhammam — APPROVED** [review]

**Pull request merged**